hswidget/palettegrideditorwidget/state.go

```diff
@@ -18,105 +17,32 @@ const (
 
 // PaletteGridState represents palette grid's state
 type widgetState struct {
-	mode widgetMode
+	Mode widgetMode `json:"mode"`
 	editEntryState
 }
 
 // Dispose cleans palette grids state
 func (ws *widgetState) Dispose() {
-	ws.mode = widgetModeGrid
-}
-
-func (ws *widgetState) Encode() []byte {
-	sw := d2datautils.CreateStreamWriter()
-
-	sw.PushInt32(int32(ws.mode))
-	sw.PushInt32(int32(ws.idx))
-	sw.PushBytes(ws.r)
-	sw.PushBytes(ws.g)
-	sw.PushBytes(ws.b)
-	sw.PushBytes(byte(len(ws.hex)))
-	sw.PushBytes([]byte(ws.hex)...)
-
-	return sw.GetBytes()
+	ws.Mode = widgetModeGrid
 }
 
 func (ws *widgetState) Decode(data []byte) {
```

Comment:
You have removed other such `Decode` methods, why are we keeping this one for the palette grid editor state?

Reply:
thanks for pointing this, I think I missed it
see c85c329